### ISSUE TITLE
RoutingKey doesn't work in Example Code

### ISSUE THREAD

**Issue by Calidus** [loekd/ServiceFabric.PubSubActors#94]
I pulled down the repo and set the routing key on the ```SampleStatelessSubscriber``` and ```and no matter what I set it to the ```HandleSampleEvent``` method always receives the message.

```
[Subscribe(routingKey: "Message=a")]
private Task HandleSampleEvent(SampleEvent sampleEvent)
{
    ServiceEventSource.Current.ServiceMessage(Context, $"Processing {sampleEvent.GetType()}: {sampleEvent.Message} on SampleStatelessSubscriber");
    return Task.CompletedTask;
}
```

I did some debugging and _```routingKeyValue``` is always null when ```ShouldDeliverMessage(MessageWrapper message)```is executed.

**Comment by danadesrosiers** [comment]
Hmm, that's not good.  I see what's happening.  `_routingKeyValue` is lost in serialization.  We should get rid of that field and make it a local variable only.  `ShouldDeliverMessage()` should use the property `RoutingKey` instead.

**Comment by Calidus** [comment]
If ```RoutingKey``` is used, ```ShouldDeliverMessage``` will have do a ```string.split``` for each message. That seems less than ideal.

**Comment by danadesrosiers** [comment]
`string.split` isn't exactly an expensive operation, but we can avoid it.  A couple options come to mind:
1. Keep the `_routingKeyValue` field and set it in `ShouldDeliverMessage` if it is not set.  At least then we don't split every time.
2. Add two properties `RoutingKeyName` and `RoutingKeyValue` to replace `RoutingKey` so we can properly serialize and deserialize the values we need without having to re-split it.

**Comment by Calidus** [comment]
I like option 2, I think it is a cleaner solution, it would keep ```ShouldDeliverMessage``` side effect free. I think the logic would become something like this:

```
if (RoutingKeyName is null || RoutingKeyName.length == 0 || RoutingKeyValue is null && RoutingKeyValue.length == 0) {
   return true;
}
/// token logic
/// return compare values
```

**Comment by danadesrosiers** [comment]
Sounds good.  Can you take care of this while you're in there adding wildcard support?

**Comment by Calidus** [comment]
Yep, should have a PR ready in a day or two. I have the basics finished, now I am trying to figure out how to add test methods.

**Comment by danadesrosiers** [comment]
Fixed in v12.2.0!
